File: 1071-greatest-common-divisor-of-strings/main.go
func gcdOfStrings(str1 string, str2 string) string {
    if str1 + str2 != str2 + str1 {
        return ""
    }

    gcdLength := gcd(len(str1), len(str2))
	return str1[:gcdLength]
}

func gcd(a, b int) int {
	for b != 0 {
		t := b
		b = a % b
		a = t
	}
	return a
}

File: 1071-greatest-common-divisor-of-strings/main.rs
impl Solution {
    pub fn gcd_of_strings(str1: String, str2: String) -> String {
        if str1.to_owned() + &str2 != str2.to_owned() + &str1 {
            return String::from("");
        }

        let gcd_length = gcd(str1.len(), str2.len());
        str1[..gcd_length].to_string()
    }
}

fn gcd(mut a: usize, mut b: usize) -> usize {

    while b != 0 {
        let temp = b;
        b = a % b;
        a = temp;
    }

    a
}
